File: aosp/src/apps/audio_equalizer/app/src/main/java/com/leandromendes/audioequalizer/data/repository/UserRepository.java
package com.leandromendes.audioequalizer.data.repository;

import com.leandromendes.audioequalizer.data.model.EqualizerProfile;

import java.util.ArrayList;
import java.util.List;

public class UserRepository {
    private final List<EqualizerProfile> equalizerProfiles = new ArrayList<>();

    public UserRepository() {
        // Initializes default profile
        equalizerProfiles.add(new EqualizerProfile());
        // TODO: Read values from the database
    }

    public void AddProfile(EqualizerProfile equalizerProfile){
        equalizerProfiles.add(equalizerProfile);
    }

    public void RemoveProfile(int index){
        equalizerProfiles.remove(index);
    }

    public void UpdateProfile(int index, EqualizerProfile equalizerProfile){
        equalizerProfiles.set(index, equalizerProfile);
    }

    public List<EqualizerProfile> getAllEqualizerProfiles() {
        return equalizerProfiles;
    }

    public EqualizerProfile getEqualizerProfile(int position){
        return equalizerProfiles.get(position);
    }

}

File: aosp/src/apps/VehicleEqualizerFull/app/src/main/java/com/leandromendes/vehicleequalizer/data/repository/UserRepository.kt
package com.leandromendes.vehicleequalizer.data.repository

import androidx.lifecycle.LiveData
import com.leandromendes.vehicleequalizer.data.ProfileDao
import com.leandromendes.vehicleequalizer.data.model.EqualizerProfile

class UserRepository(private val profileDao: ProfileDao) {
    // Returns LiveData of profiles. This list will be used by the ViewModel.
    val allEqualizerProfiles: LiveData<List<EqualizerProfile>> = profileDao.getAllProfiles()

    suspend fun addProfile(equalizerProfile: EqualizerProfile) {
        // ID is 0 (autoGenerate) so that Room inserts it as new
        profileDao.insert(equalizerProfile.copy(id = 0))
    }

    suspend fun removeProfile(profile: EqualizerProfile) {
        profileDao.delete(profile)
    }

    suspend fun updateProfile(equalizerProfile: EqualizerProfile) {
        // Assume that the EqualizerProfile object already has the database ID
        profileDao.update(equalizerProfile)
    }
}